adagucserverEC/utils/LayerMetadataStore.cpp

```diff
@@ -0,0 +1,298 @@
+#include "./LayerMetadataStore.h"
+
+#include <json_adaguc.h>
+#include <CDBFactory.h>
+#include "XMLGenUtils.h"
+
+int storeMyWMSLayerIntoMetadataDb(MetadataLayer *myWMSLayer) {
+  storeLayerMetadataStructIntoMetadataDb(myWMSLayer);
+  storeLayerDimensionListIntoMetadataDb(myWMSLayer);
+  storeLayerProjectionAndExtentListIntoMetadataDb(myWMSLayer);
+  storeLayerStyleListIntoMetadataDb(myWMSLayer);
+  return 0;
+}
+
+CT::string getLayerMetadataFromDb(MetadataLayer *myWMSLayer, CT::string metadataKey) {
+  CT::string layerName = myWMSLayer->dataSource->getLayerName();
+  CT::string datasetName = myWMSLayer->dataSource->srvParams->datasetLocation;
+  if (datasetName.empty()) {
+    // CDBDebug("Not a dataset");
+    return "";
+  }
+  return CDBFactory::getDBAdapter(myWMSLayer->dataSource->srvParams->cfg)->getLayerMetadata(datasetName, layerName, metadataKey);
+}
+
+int storeLayerMetadataInDb(MetadataLayer *myWMSLayer, CT::string metadataKey, std::string metadataBlob) {
+  try {
+    CT::string datasetName = myWMSLayer->dataSource->srvParams->datasetLocation;
```

Comment:
`srvParams` seems like a strange place to get/store the dataset name?

Reply:
srvParams is populated with items from the query string when the process runs. It is passed around to obtain information like format, datasetname, layers, etc....